cmd/launcher/flare.go

```diff
@@ -11,23 +12,47 @@ import (
 	"github.com/kolide/kit/ulid"
 	"github.com/kolide/launcher/pkg/agent/flags"
 	"github.com/kolide/launcher/pkg/agent/knapsack"
+	"github.com/kolide/launcher/pkg/agent/types"
 	"github.com/kolide/launcher/pkg/debug/checkups"
+	"github.com/kolide/launcher/pkg/debug/flareshipping"
 	"github.com/kolide/launcher/pkg/launcher"
 )
 
+// runFlareInt is an empty struct that implements the RunFlare method so we can pass to flare shipper
+type runFlareInt struct{}
+
+func (_ runFlareInt) RunFlare(ctx context.Context, k types.Knapsack, flare io.Writer, environment checkups.RuntimeEnvironmentType) error {
+	return checkups.RunFlare(ctx, k, flare, environment)
+}
+
 func runFlare(args []string) error {
 	// Flare assumes a launcher installation (at least partially) exists
 	// Overriding some of the default values allows options to be parsed making this assumption
 	// TODO this stuff needs some deeper thinking
 	launcher.DefaultAutoupdate = true
-	setDefaultPaths()
+	launcher.SetDefaultPaths()
 	_ = launcher.DefaultBinDirectoryPath
 
 	opts, err := launcher.ParseOptions("flare", args)
 	if err != nil {
 		return err
 	}
 
+	logger := log.NewLogfmtLogger(os.Stdout)
+	fcOpts := []flags.Option{flags.WithCmdLineOpts(opts)}
+	flagController := flags.NewFlagController(logger, nil, fcOpts...)
+	k := knapsack.New(nil, flagController, nil)
+
+	ctx := context.Background()
+
+	var requestUrl = env.String("KOLIDE_AGENT_FLARE_REQUEST_URL", "")
```

Comment:
don't really like determining this big a fork in logic based on an env flag ... should we make a separate command for FlareShipping or look for `--flare_upload_url=https://32PiecesOfFlair.com` or something like that?

Reply:
This is the path taken when running from the cmdline as stand alone, just starting here since it already exists. I was thinking that the flare triggered by the control server with run InSitu?